libs/core/langchain_core/utils/json_schema.py

```diff
@@ -18,8 +18,7 @@ def _retrieve_ref(path: str, schema: dict) -> dict:
         elif component.isdigit() and int(component) in out:
             out = out[int(component)]
         else:
-            # Returning the original schema or a default value if the key is not found
```

Comment:
This code was added to ensure that the agent runs even if there is a problem with the swagger provided by an external solution. We decided that it was right to remove these reference errors through preprocessing logic before the user applied swagger to the agent, so in this case, we modified the code so that a ValueError was raised and also restored the test code.